### ISSUE TITLE
The Parameter in llama-recipes/src/llama_recipes/configs/peft.py seems not be used during finetuning.

### ISSUE THREAD

**Issue by xuqianmamba** [meta-llama/llama-recipes#814]
### System Info

PyTorch version: 2.5.1+cu124
Is debug build: False
CUDA used to build PyTorch: 12.4
ROCM used to build PyTorch: N/A

OS: Ubuntu 22.04.3 LTS (x86_64)
GCC version: (Ubuntu 11.4.0-1ubuntu1~22.04) 11.4.0
Clang version: Could not collect
CMake version: version 3.22.1
Libc version: glibc-2.35

Python version: 3.10.12 (main, Nov  6 2024, 20:22:13) [GCC 11.4.0] (64-bit runtime)
Python platform: Linux-6.8.0-49-generic-x86_64-with-glibc2.35
Is CUDA available: True
CUDA runtime version: Could not collect
CUDA_MODULE_LOADING set to: LAZY
GPU models and configuration: 
GPU 0: NVIDIA GeForce RTX 4080
GPU 1: NVIDIA A100 80GB PCIe

Nvidia driver version: 560.35.05
cuDNN version: Probably one of the following:
/usr/lib/x86_64-linux-gnu/libcudnn.so.9.5.1
/usr/lib/x86_64-linux-gnu/libcudnn_adv.so.9.5.1
/usr/lib/x86_64-linux-gnu/libcudnn_cnn.so.9.5.1
/usr/lib/x86_64-linux-gnu/libcudnn_engines_precompiled.so.9.5.1
/usr/lib/x86_64-linux-gnu/libcudnn_engines_runtime_compiled.so.9.5.1
/usr/lib/x86_64-linux-gnu/libcudnn_graph.so.9.5.1
/usr/lib/x86_64-linux-gnu/libcudnn_heuristic.so.9.5.1
/usr/lib/x86_64-linux-gnu/libcudnn_ops.so.9.5.1
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: True

### Information

- [X] The official example scripts
- [ ] My own modified scripts

### 🐛 Describe the bug

I modified the parameters of llama-recipes/src/llama_recipes/configs/peft.py, including r and target_modules, and found that even tuning r to a very large number (e.g., from 8 to hidden_size) does not have an effect on the fine-tuning time.

I would like to ask how do I specify the parameters for LoRA related parameters instead, is it only by modifying this ``llama-recipes/src/llama_recipes/configs/peft.py'' file?

@dataclass
class lora_config:
     r: int=256
     lora_alpha: int=32
     target_modules: List[str] = field(default_factory=lambda: ["q_proj", "k_proj"])
     bias= "none"
     task_type: str= "CAUSAL_LM"
     lora_dropout: float=0.05
     inference_mode: bool = False

The command I used is 

(in llama-recipes/recipes/quickstart/finetuning)

CUDA_VISIBLE_DEVICES=1 FSDP_CPU_RAM_EFFICIENT_LOADING=1 python3 finetuning.py \
  --use_peft \
  --peft_method lora \
  --quantization 8bit \
  --model_name ``path to /models--meta-llama--Llama-3.2-1B/snapshots/4e20de362430cd3b72f300e6b0f18e50e7166e08'' \
  --output_dir Path/to/save/PEFT/model \
  --dataset samsum_dataset \
  --use_fp16 True \
  --batch_size_training 4

### Error logs

r=256 takes 05:53, while

peft Config: LoraConfig(task_type='CAUSAL_LM', peft_type=<PeftType.LORA: 'LORA'>, auto_mapping=None, base_model_name_or_path=None, revision=None, inference_mode=False, r=256, target_modules={'k_proj', 'q_proj'}, exclude_modules=None, lora_alpha=32, lora_dropout=0.05, fan_in_fan_out=False, bias='none', use_rslora=False, modules_to_save=None, init_lora_weights=True, layers_to_transform=None, layers_pattern=None, rank_pattern={}, alpha_pattern={}, megatron_config=None, megatron_core='megatron.core', loftq_config={}, eva_config=None, use_dora=False, layer_replication=None, runtime_config=LoraRuntimeConfig(ephemeral_gpu_offload=False), lora_bias=False)
trainable params: 73,400,320 || all params: 3,286,150,144 || trainable%: 2.2336
Using the latest cached version of the module from /home/xuqian/.cache/huggingface/modules/datasets_modules/datasets/Samsung--samsum/f1d7c6b7353e6de335d444e424dc002ef70d1277109031327bc9cc6af5d3d46e (last modified on Tue Dec 10 19:20:15 2024) since it couldn't be found locally at Samsung/samsum, or remotely on the Hugging Face Hub.
--> Training Set Length = 14732
Using the latest cached version of the module from /home/xuqian/.cache/huggingface/modules/datasets_modules/datasets/Samsung--samsum/f1d7c6b7353e6de335d444e424dc002ef70d1277109031327bc9cc6af5d3d46e (last modified on Tue Dec 10 19:20:15 2024) since it couldn't be found locally at Samsung/samsum, or remotely on the Hugging Face Hub.
--> Validation Set Length = 818
Preprocessing dataset: 100%|█████████████████████████████████████████████████████████████████████████████████████████| 14732/14732 [00:03<00:00, 3870.16it/s]
length of dataset_train 639
--> Num of Training Set Batches loaded = 319
Preprocessing dataset: 100%|█████████████████████████████████████████████████████████████████████████████████████████████| 818/818 [00:00<00:00, 4086.63it/s]
--> Num of Validation Set Batches loaded = 34
--> Num of Validation Set Batches loaded = 34
/home/xuqian/LLM_security/llama-recipes/src/llama_recipes/utils/train_utils.py:93: FutureWarning: `torch.cuda.amp.GradScaler(args...)` is deprecated. Please use `torch.amp.GradScaler('cuda', args...)` instead.
  scaler = torch.cuda.amp.GradScaler()
Starting epoch 0/3
train_config.max_train_step: 0
/home/xuqian/.local/lib/python3.10/site-packages/torch/cuda/memory.py:365: FutureWarning: torch.cuda.reset_max_memory_allocated now calls torch.cuda.reset_peak_memory_stats, which resets /all/ peak memory stats.
  warnings.warn(
Training Epoch: 1:   0%|                                                                                                             | 0/319 [00:00<?, ?it/s]/home/xuqian/LLM_security/llama-recipes/src/llama_recipes/utils/train_utils.py:153: FutureWarning: `torch.cuda.amp.autocast(args...)` is deprecated. Please use `torch.amp.autocast('cuda', args...)` instead.
  with autocast():
Training Epoch: 1/3, step 318/319 completed (loss: 1.2523458003997803): 100%|██████████████████████████████████████████████| 319/319 [05:53<00:00,  1.11s/it]

### Expected behavior

r=256 and 8 takes different time.

**Comment by xuqianmamba** [comment]
r=8 takes 05:48

peft Config: LoraConfig(task_type='CAUSAL_LM', peft_type=<PeftType.LORA: 'LORA'>, auto_mapping=None, base_model_name_or_path=None, revision=None, inference_mode=False, r=8, target_modules={'k_proj', 'q_proj'}, exclude_modules=None, lora_alpha=32, lora_dropout=0.05, fan_in_fan_out=False, bias='none', use_rslora=False, modules_to_save=None, init_lora_weights=True, layers_to_transform=None, layers_pattern=None, rank_pattern={}, alpha_pattern={}, megatron_config=None, megatron_core='megatron.core', loftq_config={}, eva_config=None, use_dora=False, layer_replication=None, runtime_config=LoraRuntimeConfig(ephemeral_gpu_offload=False), lora_bias=False)
trainable params: 2,293,760 || all params: 3,215,043,584 || trainable%: 0.0713
--> Training Set Length = 14732
--> Validation Set Length = 818
Preprocessing dataset: 100%|████████████████████████████████████████████████████████████| 14732/14732 [00:03<00:00, 3718.54it/s]
length of dataset_train 639
--> Num of Training Set Batches loaded = 319
Preprocessing dataset: 100%|████████████████████████████████████████████████████████████████| 818/818 [00:00<00:00, 3971.97it/s]
--> Num of Validation Set Batches loaded = 34
--> Num of Validation Set Batches loaded = 34
/home/xuqian/LLM_security/llama-recipes/src/llama_recipes/utils/train_utils.py:93: FutureWarning: `torch.cuda.amp.GradScaler(args...)` is deprecated. Please use `torch.amp.GradScaler('cuda', args...)` instead.
  scaler = torch.cuda.amp.GradScaler()
Starting epoch 0/3
train_config.max_train_step: 0
/home/xuqian/.local/lib/python3.10/site-packages/torch/cuda/memory.py:365: FutureWarning: torch.cuda.reset_max_memory_allocated now calls torch.cuda.reset_peak_memory_stats, which resets /all/ peak memory stats.
  warnings.warn(
Training Epoch: 1:   0%|                                                                                | 0/319 [00:00<?, ?it/s]/home/xuqian/LLM_security/llama-recipes/src/llama_recipes/utils/train_utils.py:153: FutureWarning: `torch.cuda.amp.autocast(args...)` is deprecated. Please use `torch.amp.autocast('cuda', args...)` instead.
  with autocast():
Training Epoch: 1/3, step 63/319 completed (loss: 1.2540254592                                                              Training Epoch: 1/3, step 318/319 completed (loss: 1.253248333
Max CUDA memory allocated was 55 GB
Max CUDA memory reserved was 60 GB
Peak active CUDA memory was 55 GB
CUDA Malloc retries : 0
CPU Total Peak Memory consumed during the train (max): 3 GB
evaluating Epoch: 100%|███████| 34/34 [00:10<00:00,  3.15it/s]
 eval_ppl=tensor(3.3337, device='cuda:0') eval_epoch_loss=tensor(1.2041, device='cuda:0')
we are about to save the PEFT modules
PEFT modules are saved in Path/to/save/PEFT/model directory
best eval loss on epoch 1 is 1.2040690183639526
Epoch 1: train_perplexity=3.6197, train_epoch_loss=1.2864, epoch time 350.48221955401823s
Starting epoch 1/3
train_config.max_train_step: 0
Training Epoch: 2/3, step 318/319 completed (loss: 1.2103567123413086): 100%|██████████████| 319/319 [05:48<00:00,  1.09s/it]
Max CUDA memory allocated was 55 GB
Max CUDA memory reserved was 61 GB
Peak active CUDA memory was 55 GB
CUDA Malloc retries : 0
